src/com/palmergames/bukkit/towny/command/NationCommand.java

```diff
@@ -1350,7 +1352,20 @@ public void mergeNation(Player player, String name) throws TownyException {
 				throw new TownyException(String.format(TownySettings.getLangString("msg_err_king_of_that_nation_is_not_online"), name, king.getName()));
 			}
 			TownyMessaging.sendMessage(BukkitTools.getPlayer(king.getName()), String.format(TownySettings.getLangString("msg_would_you_merge_your_nation_into_other_nation"), nation, remainingNation, remainingNation));
-			ConfirmationHandler.addConfirmation(king, ConfirmationType.NATION_MERGE, remainingNation);
+			Confirmation confirmation = new Confirmation(player);
+			Nation finalRemainingNation = remainingNation;
+			confirmation.setHandler(() -> {
+				try {
+					Nation succumbingNation = resident.getTown().getNation();
+					Nation prevailingNation = finalRemainingNation;

```

Comment:
Can we not reuse the above nation and remainingNation here now?

Also finalRemainingNation is basically unused.